pkg/imgpkg/signature/cosign.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package signature
+
+import (
+	"fmt"
+	"net/http"
+
+	regname "github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote/transport"
+	"github.com/k14s/imgpkg/pkg/imgpkg/imageset"
+	"github.com/k14s/imgpkg/pkg/imgpkg/registry"
+	"github.com/k14s/imgpkg/pkg/imgpkg/signature/cosign"
+)
+
+func NewCosign(reg registry.Registry) *Cosign {
+	return &Cosign{registry: reg}
+}
+
+type Cosign struct {
+	registry registry.Registry
+}
+
+func (c Cosign) Signature(imageRef regname.Digest) (imageset.UnprocessedImageRef, error) {
+	sigTagRef, err := c.signatureTag(imageRef)
+	if err != nil {
+		return imageset.UnprocessedImageRef{}, err
+	}
+
+	signature, err := c.registry.Digest(sigTagRef)
+	if err != nil {
+		if transportErr, ok := err.(*transport.Error); ok {
+			if transportErr.StatusCode == http.StatusNotFound {
+				return imageset.UnprocessedImageRef{}, NotFound{}
+			}
+		}
+		return imageset.UnprocessedImageRef{}, err
+	}
+
+	signDigest := imageRef.Digest(signature.String())
```

Comment:
i dont think we need this line. sigTagRef is already relative to imageRef and above registry.Digest call will get the digest ref.

Reply:
We use this call to see if the image exist or not, maybe we should have a function that wrap the Digest call